### PR TITLE
GSoC2023: added Gnuastro's project

### PR DESCRIPTION
**Pull request by mohammad-akhlaghi**

### CODE REVIEW

**Review by mirca — APPROVED** [review]

File: _projects/2023/gnuastro/python_wrappers.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+However, Gnuastro's also has an extensive set of installed, dynamic C/C++ libraries, which are actually the beating heart of all the programs (they do all the number crunching).
+On the other hand, many projects today are done in Python, almost excuslively using Numpy for their numerical operations (like data arrays).
+Python and Numpy are actually written in C, therefore they have very well-defined interfaces for communicating with installed C libraries.
+For example see [the core C library of Numpy](https://numpy.org/doc/stable/reference/c-api/index.html).
+
+As part of GSoC 2022, Jash Shah completed a first implementation of pyGnuastro, see https://github.com/Jash-Shah/pyGnuastro and https://test.pypi.org/project/pygnuastro.
+This was mostly focused on the low-level build system (for example to be integrated nicely with PyPi).
+With the inclusion of more libraries we discovered that Gnuastro's error reporting also [needs modification](https://savannah.gnu.org/task/?16263) for blending into Python.
+Jash also progressed nicely on this front in GSoC 2022, but it is not yet finished.
```
Comment (bsipocz) [inline _projects/2023/gnuastro/python_wrappers.md:42]: I would rephrase this to tune back from the progress/final report style for last year's project and more focus on what's needed to be done this year.

File: _projects/2023/gnuastro/python_wrappers.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+This project therefore has two components: 1. completing the new error-reporting features of Gnuastro in its own Library. 2. Bringing the Gnuastro libraries into the pyGnuastro.
+This will be a wonderful chance to master Python and Scipy/Numpy at a very fundamental level, giving you a great experience to expand what you have learnt afterwards into many other functionalities (far beyond astronomy!).
+Of course, in the meantime you will also working on many real-world astronomical data and science scenarios, while collaborating with an international team using ground-based and space-based data.
```
Comment (bsipocz) [inline _projects/2023/gnuastro/python_wrappers.md:46]: > many real-world astronomical data and science scenarios

if there will be many of these, list a few concrete examples instead.

File: _projects/2023/gnuastro/python_wrappers.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+# Student requirements:
+ - Low-level Python and Numpy (their C library).
+difficulty: low
+issues:
+# Related issues (if any) to this project.
+ - https://github.com/Jash-Shah/pyGnuastro
+mentors:
+# First person in contact; mentors may change before project starts.
+# GitHub handles
+ - mohammad-akhlaghi
```
Comment (bsipocz) [inline _projects/2023/gnuastro/python_wrappers.md:14]: as always, it would be great to have someone would commit being a second mentor.

**Review by bsipocz — APPROVED** [review]
I think we should get this in as is for now, but I left a few suggestions for rephrasing some of the details by the time students are showing up.
